### PR TITLE
fix(plugin-messages): update message created event handling

### PR DESCRIPTION
**Pull request by InteractiveTimmy**
# Pull Request

## Description

Update how message:created events are shaped so that they properly
populate the text, markdown, and files fields. Clean up testing
so that it handles this properly, and adjust testing statements
to utilize the proper 'files' field instead of 'file'. Update
documentation for messages.listen() to identify how to retrieve
the html field data to satisfy the related internal case.

Fixes https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-78472

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Utilized repository testing suite on the associated module and project.

**Test Configuration**:
* Node/Browser Version v8.16.0
* NPM Version v6.4.1

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/node_modules/@webex/plugin-messages/src/messages.js [hunk 1]
```diff
@@ -54,6 +54,11 @@ const Messages = WebexPlugin.extend({
    * This is an alternate approach to registering for messages webhooks.
    * The events passed to any registered handlers will be similar to the webhook JSON,
    * but will omit webhook specific fields such as name, secret, url, etc.
+   * The messages.listen() event objects can also include additional fields not
+   * available in webhook JSON. These fields are text, markdown, and files.
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-messages/src/messages.js]: ...in the webhook's JSON payload: \`text\`, \`markdown\`, and \`files\`.

File: packages/node_modules/@webex/plugin-messages/src/messages.js [hunk 2]
```diff
@@ -54,6 +54,11 @@ const Messages = WebexPlugin.extend({
...
+   * These fields are available when their details are included in the WebSocket's
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-messages/src/messages.js]: web socket's

File: packages/node_modules/@webex/plugin-messages/src/messages.js [hunk 3]
```diff
@@ -54,6 +54,11 @@ const Messages = WebexPlugin.extend({
...
+   * activity object. Retrieving other fields, such as the html field,
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-messages/src/messages.js]: \`activity\` object. Retrieving other fields, such as the \`html\` field,

File: packages/node_modules/@webex/plugin-messages/src/messages.js [hunk 4]
```diff
@@ -351,10 +356,16 @@ const Messages = WebexPlugin.extend({
         activity.actor.emailAddress || activity.actor.entryEmail;
 
       if (event !== SDK_EVENT.EXTERNAL.EVENT_TYPE.DELETED) {
+        const {content, displayName} = activity.object;
         const files = getHydraFiles(activity);
 
         sdkEvent.data.id = constructHydraId(hydraTypes.MESSAGE, activity.id);
-        sdkEvent.data.text = activity.object.displayName;
+        if (content || displayName) {
+          sdkEvent.data.text = content || displayName;
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-messages/src/messages.js]: If this check is needed, you could simplify with: 

```
const text = content || displayname;
if (text) {
...
```

File: packages/node_modules/@webex/plugin-messages/src/messages.js (hunk 1 above)
Comment (InteractiveTimmy) [reply packages/node_modules/@webex/plugin-messages/src/messages.js]: I have addressed this change.

File: packages/node_modules/@webex/plugin-messages/src/messages.js (hunk 2 above)
Comment (InteractiveTimmy) [reply packages/node_modules/@webex/plugin-messages/src/messages.js]: I have addressed this change.

File: packages/node_modules/@webex/plugin-messages/src/messages.js (hunk 3 above)
Comment (InteractiveTimmy) [reply packages/node_modules/@webex/plugin-messages/src/messages.js]: I have addressed this change.

File: packages/node_modules/@webex/plugin-messages/src/messages.js (hunk 4 above)
Comment (InteractiveTimmy) [reply packages/node_modules/@webex/plugin-messages/src/messages.js]: I have addressed this change.

**Pull request merged**